src/Scale.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import classNames from 'classnames';
+import warning from 'warning';
+
+function calcPoints(marks, dots, step, min, max) {
+  warning(
+    dots ? step > 0 : true,
+    '`Slider[step]` should be a positive number in order to make Slider[dots] work.',
+  );
+  const points = Object.keys(marks).map(parseFloat);
+  if (dots) {
+    for (let i = min; i <= max; i += step) {
+      if (!(points.indexOf(i) >= 0)) {
+        points.push(i);
+      }
+    }
+  }
+  return points;
+}
+
+const createMarks = (className, marks, included, upperBound, lowerBound) => {
+  const marksKeys = Object.keys(marks);
+
+  const elements = marksKeys.map(parseFloat).sort((a, b) => a - b).map((point) => {
+    const isActived = (!included && point === upperBound) ||
+            (included && point <= upperBound && point >= lowerBound);
+    const markClassName = classNames({
+      [`${className}-text`]: true,
+      [`${className}-text-active`]: isActived,
+    });
+
+    const markPoint = marks[point];
+    const markPointIsObject = typeof markPoint === 'object' &&
+            !React.isValidElement(markPoint);
+    const markLabel = markPointIsObject ? markPoint.label : markPoint;
+    return (
+      <span className={markClassName} key={point}>
+        {markLabel}
+      </span>);
+  });
+
+  return elements;
+};
+
+const Scale = ({ prefixCls, marks, dots, step, included, handles,
+                tracks, lowerBound, upperBound, max, min }) => {
+  const dotNodes = calcPoints(marks, dots, step, min, max).map((point) => {
```

Comment:
Sometimes you call them "nodes", sometimes "elements". I would always use "elements".